streams/authorizer/src/main/java/com/hortonworks/streamline/streams/security/authentication/StreamlineKerberosRequestFilter.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2017 Hortonworks.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package com.hortonworks.streamline.streams.security.authentication;
+
+import com.hortonworks.streamline.common.exception.service.exception.request.WebserviceAuthorizationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.SecurityContext;
+import java.io.IOException;
+import java.security.Principal;
+
+public class StreamlineKerberosRequestFilter implements ContainerRequestFilter {
```

Comment:
Is there a reason for adding this as a ContainerRequestFilter rather than a ServletFilter. We already have a method called addServletFilters in StreamlineApplication and it picks up all filters configured in streamline.yaml